### PR TITLE
ESM notification timeout

### PR DESCRIPTION
**Pull request by NBerkel**
Introduces support for notification timeout: removes ESM notification and clears ESM queue after set timeout.

### CODE REVIEW

File: aware-core/src/main/java/com/aware/ui/ESM_Queue.java [hunk 1]
```diff
@@ -33,7 +36,7 @@
 
     public ESM_State esmStateListener = new ESM_State();
 
-    private ESMFactory esmFactory = new ESMFactory();
+    private static ESMFactory esmFactory = new ESMFactory();
```
Comment (denzilferreira) [inline aware-core/src/main/java/com/aware/ui/ESM_Queue.java]: Does this need to be static?

File: aware-core/src/main/java/com/aware/ui/ESM_Queue.java [hunk 2]
```diff
@@ -52,13 +55,16 @@ protected void onCreate(Bundle bundle) {
...
         filter.addAction(ESM.ACTION_AWARE_ESM_EXPIRED);
+        filter.addAction(ESM.ACTION_AWARE_ESM_TIMEOUT);
         registerReceiver(esmStateListener, filter);
     }
 
     @Override
     protected void onResume() {
         super.onResume();
 
+        Log.d("Niels", "onResume called: queue=" + ESM_Queue.getQueueSize(this));
```
Comment (denzilferreira) [inline aware-core/src/main/java/com/aware/ui/ESM_Queue.java]: Remove this

File: aware-core/src/main/java/com/aware/ui/ESM_Queue.java [hunk 3]
```diff
@@ -95,7 +101,13 @@ protected void onResume() {
         @Override
         public void onReceive(Context context, Intent intent) {
             if (intent.getAction().equals(ESM.ACTION_AWARE_ESM_QUEUE_COMPLETE)) {
+                Log.d("NIELS", "Complete intent caught!");
```
Comment (denzilferreira) [inline aware-core/src/main/java/com/aware/ui/ESM_Queue.java]: Remove

File: aware-core/src/main/java/com/aware/ui/esms/ESM_Question.java [hunk 4]
```diff
@@ -357,6 +380,24 @@ private void dismissESM() {
         if (esm_dialog != null) esm_dialog.dismiss();
     }
 
+    /**
+     * When one of the ESM's has timed out, the entire queue gets removed.
+     */
+    public void removeESM(Context context) {
```
Comment (denzilferreira) [inline aware-core/src/main/java/com/aware/ui/esms/ESM_Question.java]: This function name may lead the user to think it only removes one ESM. Rename to timeoutQueue

File: aware-tests/src/main/AndroidManifest.xml [hunk 5]
```diff
@@ -8,6 +8,8 @@
         android:supportsRtl="true"
         android:theme="@style/AppTheme">
 
+        <uses-permission android:name="android.permission.WRITE_EXTERNAL_STORAGE" />
```
Comment (denzilferreira) [inline aware-tests/src/main/AndroidManifest.xml]: Not needed.

File: aware-tests/src/main/java/com/aware/tests/TestESM.java [hunk 6]
```diff
@@ -22,9 +26,9 @@
...
     @Override
     public void test(Context context) {
-//        testESMS(context);
+        testESMS(context);
 //        trialESMS(context);
-        testFlow(context);
+//        testFlow(context);
     }
 
     private void testFlow(Context context) {
```
Comment (denzilferreira) [inline aware-tests/src/main/java/com/aware/tests/TestESM.java]: These tests were for flow functionality. Create and use testNotifications() to not mix tests.

Comment (NBerkel) [inline aware-tests/src/main/java/com/aware/tests/TestESM.java]: Separated into two tests.

**Pull request merged**